specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json

```diff
@@ -0,0 +1,272 @@
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "InstancePoolOperationListResult": {
+      "description": "The response to a list instance pool operations request",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InstancePoolOperation"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "InstancePoolOperationProperties": {
+      "description": "The properties of a instance pool operation.",
+      "type": "object",
+      "properties": {
+        "instancePoolName": {
+          "description": "The name of the instance pool the operation is being performed on.",
+          "type": "string",
+          "readOnly": true
+        },
+        "operation": {
+          "description": "The name of operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "operationFriendlyName": {
+          "description": "The friendly name of operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "percentComplete": {
+          "format": "int32",
+          "description": "The percentage of the operation completed.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "startTime": {
+          "format": "date-time",
+          "description": "The operation start time.",
+          "type": "string",
+          "readOnly": true
+        },
+        "state": {
+          "description": "The operation state.",
+          "enum": [
+            "Pending",
+            "InProgress",
+            "Succeeded",
+            "Failed",
+            "CancelInProgress",
+            "Cancelled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ManagementOperationState",
+            "modelAsString": true
+          }
+        },
+        "errorCode": {
+          "format": "int32",
+          "description": "The operation error code.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "errorDescription": {
+          "description": "The operation error description.",
+          "type": "string",
+          "readOnly": true
+        },
+        "errorSeverity": {
+          "format": "int32",
+          "description": "The operation error severity.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "errorType": {
+          "description": "Error type (e.g. None, User).",
+          "enum": [
+            "None",
+            "User"
+          ],
+          "type": "string",
+          "readOnly": true
+        },
```

Comment:
why are these needed in properties ? shouldn't they online come in error responses. and there is an error response definition in common-types